public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,903 @@
+include ../_util-fns
+
+:marked
+  _Reactive forms_ is an Angular technique for creating forms in a _reactive_ style.
+  This guide explains reactive forms as you follow the steps to build a "Hero Detail Editor" form.
+
+a#toc
+:marked
+  ## Contents
+
+  - [Introduction to reactive forms](#intro)
+  - [Setup](#setup)
+  - [Create a data model](#data-model)
+  - [Create a _reactive forms_ component](#create-component)
+  - [Create its template file](#create-template)
+  - [Import the _ReactiveFormsModule_](#import)
+  - [Update the app to use the form component](#update)
+  - [Introduction to _FormBuilder_](#formbuilder)
+  - [Nested FormGroups](#grouping)
+  - [Inspect _FormControl_ properties](#properties)
+  - [Set form model data using _setValue_ and _patchValue_](#set-data)
+  - [Use _FormArray_ to present an array of _FormGroups_](#form-array)
+  - [Observe control changes](#observe-control)
+  - [Save form data](#save)
+
+  Try the <live-example plnkr="final" title="Reactive Forms (final) in Plunker">Reactive Forms live-example</live-example>.
+
+  You can also run the <live-example title="Reactive Forms Demo in Plunker">Reactive Forms Demo</live-example> version
+  and choose one of the intermediate steps from the "demo picker" at the top.
+
+a#intro
+:marked
+  ## Introduction to Reactive Forms
+
+  Angular offers two form-building technologies: _reactive_ forms and _template-driven_ forms.
+  The two technologies belong to the `@angular/forms` library
+  and share a common set of form control classes.
+
+  But they diverge markedly in philosophy, programming style, and technique.
+  They even have their own modules: the `ReactiveFormsModule` and the `FormsModule`.
+
+  ### _Reactive_ forms
+  Angular _reactive_ forms facilitate a _reactive style_ of programming 
+  that favors explicit management of the data flowing between
+  a non-UI _data model_ (typically retrieved from a server) and a 
+  UI-oriented _form model_ that retains the states 
+  and values of the HTML controls on screen.
+
+  With _reactive_ forms, you create a tree of Angular form control objects
+  in the component class and bind them to native form control elements in the 
+  component template, using techniques described in this guide.
+
+  With reactive forms, you create and manipulate form control objects directly in the 
+  component class. As the component class has immediate access to both the data 
+  model and the form control structure, you can push data model values into 
+  the form controls and pull user-changed values back out. The component can 
+  observe changes in form control state and react to those changes.
+
+  One advantage of working with form control objects directly is that value and validity updates 
+  are always synchronous and under your control. 
+  You won't encounter the timing issues that sometimes plague a template-driven form.
```

Comment:
I think we still need to mention that template-driven forms are async, or this point doesn't make sense.